services/src/main/java/com/linkedin/kmf/common/Utils.java

```diff
@@ -35,13 +54,178 @@
+      topicConfig.setProperty(KafkaConfig.MinInSyncReplicasProp(), Integer.toString(minIsr));
+      AdminUtils.createTopic(zkUtils, topic, partitionCount, replicationFactor, new Properties());
+
+      LOG.info("Created monitoring topic \"" + topic + "\" with " + partitionCount + " partitions, min ISR of " + minIsr
+          + " and replication factor of " + replicationFactor + ".");
+
+      return partitionCount;
+    } finally {
+      zkUtils.close();
+    }
+  }
+
+  /**
+   * This is so you can call KafkaConsumer.partitionsFor don't consume messages from this that is the job of the
+   * ConsumeService.
+   *
+   * @param brokerList probably get this from the configuration file.
+   * @param zkUrl zoo keeper url
+   * @param topic the monitoring topic
+   * @return an initialized KafkaConsumer
+   */
+
+  private static KafkaConsumer<?,?> constructClientForMetadata(String brokerList, String zkUrl, String topic) {
+
+    Properties consumerProps = new Properties();
+
+    // Assign default config. This has the lowest priority.
+    consumerProps.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+    consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
+    consumerProps.put(ConsumerConfig.CLIENT_ID_CONFIG, "kmf-metadata-consumer");
+    consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "kmf-metadata-consumer-group");
+    consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+    consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+
+    // Assign config specified for ConsumeService.
+    consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
+    consumerProps.put("zookeeper.connect", zkUrl);
+
+    KafkaConsumer<String, String> metadataConsumer = new KafkaConsumer<>(consumerProps);
+    metadataConsumer.subscribe(Collections.singletonList(topic));
+    return metadataConsumer;
+  }
+
+  /**
+   *
+   * @param zkUrl the zookeeper url
+   * @param topic the monitored topic
+   * @param partitionThreshold  This assumes that you want to have at least one per broker.  When the number of partitions
+   * @param brokersList this is the list of brokers we would need to create a consumer connection.
+   * @return true if the monitored topic is no longer distributed evenly with respect to the given parameters
+   */
+  public static boolean monitoredTopicNeedsRebalance(String zkUrl, String topic, String brokersList,
+      double partitionThreshold) {
+
+    if (partitionThreshold < 1) {
+      throw new IllegalArgumentException(
+          "partitionThreshold must be greater than or equal to one, but found " + partitionThreshold + ".");
+    }
+
+    ZkUtils zkUtils = ZkUtils.apply(zkUrl, ZK_SESSION_TIMEOUT_MS, ZK_CONNECTION_TIMEOUT_MS, JaasUtils.isZkSecurityEnabled());
+
+    try (KafkaConsumer<?, ?> consumerForMetadata = constructClientForMetadata(brokersList, zkUrl, topic)) {
```

Comment:
I started down that route but in involves converting a scala data structure into java.  It seems somewhat complicated since it involves nested collections.  Specifically Map[Int, Seq[Int]] needs to be converted.  In the past I've had problems with using java.lang.Integer in collections when calling java methods that have arguments like Collection<Integer>.  If you know how to do this conversion and it's not going to take all day to figure it out then I'm open to using the equivalent functionality in ZkUtils.